client/src/components/DataFiles/DataFilesDropdown/DataFilesDropdown.jsx

```diff
@@ -0,0 +1,115 @@
+  DropdownMenu,
+  DropdownItem,
+  ButtonDropdown,
+} from 'reactstrap';
+import { useSystemDisplayName, useSystems } from 'hooks/datafiles';
+import '../DataFilesBreadcrumbs/DataFilesBreadcrumbs.scss';
+
+const BreadcrumbsDropdown = ({
+  api,
+  scheme,
+  system,
+  path,
+  section,
+  isPublic,
+}) => {
+  const paths = [];
+  const [dropdownOpen, setDropdownOpen] = useState(false);
+  const toggleDropdown = () => setDropdownOpen(!dropdownOpen);
+
+  const handleNavigation = (targetPath) => {
+    const basePath = isPublic ? '/public-data' : '/workbench/data';
+    let url;
+
+    if (scheme === 'projects' && !targetPath) {
+      url = `${basePath}/${api}/projects/`;
+    } else if (api === 'googledrive' && !targetPath) {
+      url = `${basePath}/${api}/${scheme}/${system}/`;
+    } else {
+      url = `${basePath}/${api}/${scheme}/${system}${targetPath}/`;
+    }
+
+    return url;
+  };
+
+  const { fetchSelectedSystem } = useSystems();
+  const selectedSystem = fetchSelectedSystem({ scheme, system, path });
+  const systemName = useSystemDisplayName({ scheme, system, path });
+  const homeDir = selectedSystem?.homeDir;
+  const isSystemRootPath = !path
+    .replace(/^\/+/, '')
+    .startsWith(homeDir?.replace(/^\/+/, ''));
+  const startingPath = isSystemRootPath ? '' : homeDir;
+
+  const pathComponents = path.split('/').filter((x) => !!x);
+  const startingPathComponents = startingPath.split('/').filter((x) => !!x);
+  const overlapIndex = pathComponents.findIndex(
+    (component, index) => startingPathComponents[index] !== component
+  );
+
+  let currentPath = startingPath;
+  pathComponents.slice(overlapIndex).forEach((component) => {
+    currentPath = `${currentPath}/${component}`;
+    paths.push(currentPath);
+  });
+
+  const fullPath = paths.reverse();
+  return (
+    <div id="path-button-wrapper">
+      <ButtonDropdown
+        isOpen={dropdownOpen}
+        toggle={toggleDropdown}
+        id="go-to-button-dropdown"
+        className="go-to-button-dropdown"
+      >
+        <DropdownToggle tag={Button}>Go to ...</DropdownToggle>
+        <DropdownMenu>
+          {fullPath.slice(1, fullPath.length).map((path, index) => {
+            const folderName = path.split('/').pop();
+            return (
+              <Link key={index} to={handleNavigation(path)}>
+                <DropdownItem>
+                  <i className="icon-folder" />
+                  {folderName.length > 20
+                    ? folderName.substring(0, 20)
+                    : folderName}
+                </DropdownItem>
+              </Link>
+            );
+          })}
+          <DropdownItem divider />
+          <Link className="link-hover" to={handleNavigation(homeDir)}>
+            <DropdownItem className="complex-dropdown-item-root">
+              <i className="icon-my-data" />
+              <span className="multiline-menu-item-wrapper">
+                {systemName || 'Shared Workspaces'}
+                <small>Root</small>
```

Comment:
Should we only define the dropdown item as "root" if it's the base path of the data files entry, i.e. `/work/1234/sal`? As it is, if you go further "up" the tree and go to https://cep.test/workbench/data/tapis/private/frontera/, this dir is now "root"
<img width="439" alt="Screenshot 2023-10-18 at 12 06 37 PM" src="https://github.com/TACC/Core-Portal/assets/20326896/86573339-b844-45d7-8116-86f668dafd29">

Additionally, if you click this "root", you get an "undefined" path appended:
<img width="465" alt="Screenshot 2023-10-18 at 12 06 32 PM" src="https://github.com/TACC/Core-Portal/assets/20326896/1bce5b8d-15de-4dda-9633-232be6a62483">

Reply:
![Screenshot 2023-10-23 at 10 32 41 AM](https://github.com/TACC/Core-Portal/assets/50084480/3e0a81bf-e145-4a24-acc0-6cc82e57280f)
With Root (if homeDir is present)
![Screenshot 2023-10-23 at 10 34 32 AM](https://github.com/TACC/Core-Portal/assets/50084480/ad8f2c57-dbfa-44d8-9a11-47d4dcbe2e26)
Without Root (if higher up in tree hierarchy, like in Frontera or Cloud.data)